packages/python/plotly/plotly/express/_core.py

```diff
@@ -1432,23 +1432,11 @@ def process_dataframe_hierarchy(args):
     _check_dataframe_all_leaves(df[path[::-1]])
     discrete_color = False
 
-    if args["color"] and args["color"] in path:
-        series_to_copy = df[args["color"]]
-        new_col_name = args["color"] + "additional_col_for_color"
-        path = [new_col_name if x == args["color"] else x for x in path]
+    for col_name in path:
+        series_to_copy = df[col_name]
+        new_col_name = col_name + "_path_copy"
+        path = [new_col_name if x == col_name else x for x in path]
```

Comment:
the new approach is just to make a full copy of everything in `path` even if they're not in `color` or whatever and not touch any of the other args, instead of spot-checking `color` and `hover_data`... Maybe there's a case here I'm missing but it's not in the tests and this approach seems more future-friendly.

Reply:
probably we don't need the `if x == col_name` here? Since `x` is in `path`.

Reply:
it's an elegant solution. I had not thought that column names in `path` were disposable since they would not appear in the final hovertemplate.

Reply:
you're right that comprehension is weird... check out the new version?